src/main/resources/typescriptc.js

```diff
@@ -266,6 +294,17 @@
             };
             output.results.push(result);
         });
+
+        // If we are outputting a single file create the array of written files
+        //  and map it against all input files
+        if (options.out) {
+            var filesWritten = createFilesWrittenArray(options.out);
```

Comment:
Can you explain how this is different than line 286?  I don't understand why you're applying this to all of the output.results here.  

As an alternative, could line 286 be the following?
var filesWritten = (options.out)? createFilesWrittenArray(options.out) : createFilesWrittenArray(outputFile);